From 5c4b642141eb3576213ab59d06ba12e1fa989046 Mon Sep 17 00:00:00 2001
From: Ronnie Sahlberg <lsahlber@redhat.com>
Date: Mon, 14 Dec 2020 16:40:27 +1000
Subject: [PATCH 173/197] cifs: fix uninitialized variable in
 smb3_fs_context_parse_param

Addresses an issue noted by the kernel test robot

Reported-by: kernel test robot <lkp@intel.com>
Signed-off-by: Steve French <stfrench@microsoft.com>
Signed-off-by: Ronnie Sahlberg <lsahlber@redhat.com>
---
 fs/cifs/fs_context.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/fs/cifs/fs_context.c b/fs/cifs/fs_context.c
index a309e494577c..ca745fa6c4a1 100644
--- a/fs/cifs/fs_context.c
+++ b/fs/cifs/fs_context.c
@@ -726,8 +726,10 @@ static int smb3_fs_context_parse_param(struct fs_context *fc,
 	 * we will need special handling of them.
 	 */
 	if (param->type == fs_value_is_string && param->string[0] == 0) {
-		if (!strcmp("pass", param->key) || !strcmp("password", param->key))
+		if (!strcmp("pass", param->key) || !strcmp("password", param->key)) {
 			skip_parsing = true;
+			opt = Opt_pass;
+		}
 	}
 
 	if (!skip_parsing) {
-- 
2.27.0

